File: payroll-api/src/main/java/com/company/payroll/mapper/AccountMapper.java
package com.company.payroll.mapper;

import java.util.List;

import com.company.payroll.model.SystemAccount;
import org.springframework.stereotype.Repository;

@Repository
public interface AccountMapper {
    int deleteByPrimaryKey(Integer aId);
    int insert(SystemAccount row);
    int insertSelective(SystemAccount row);
    List<SystemAccount> selectList();
    SystemAccount selectByPrimaryKey(Integer aId);
    SystemAccount selectByUsername(String username);
    int updateByPrimaryKeySelective(SystemAccount row);
    int updateImagePathByUsername(SystemAccount row);
    int updatePasswordByUsername(SystemAccount row);
    int updateByPrimaryKey(SystemAccount row);
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/mapper/AccountMapper.kt
package com.company.payroll.mapper

import com.company.payroll.model.SystemAccount
import org.springframework.stereotype.Repository

@Repository
interface AccountMapper {
  fun deleteByPrimaryKey(aId: Int): Int

  fun insert(row: SystemAccount): Int

  fun insertSelective(row: SystemAccount): Int

  fun selectList(): List<SystemAccount>

  fun selectByPrimaryKey(aId: Int): SystemAccount

  fun selectByUsername(username: String): SystemAccount

  fun updateByPrimaryKeySelective(row: SystemAccount): Int

  fun updateImagePathByUsername(row: SystemAccount): Int

  fun updatePasswordByUsername(row: SystemAccount): Int

  fun updateByPrimaryKey(row: SystemAccount): Int
}